src/hooks/useEmailDisplayError.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {useState} from 'react';
+import {Text, TextProps} from 'react-native';
+import {Styles} from '../screens/UseEmailDisplayErrorHookScreen/styles';
+
+type TextElement = JSX.Element & {type: typeof Text} & TextProps;
+
+type useEmailOutput = {
+  email: string;
+  emailError: undefined | string;
+  setEmail: (emailInput: string) => void;
+  errorTextComponent: TextElement;
+};
+
+const emailRegExp = new RegExp('^[a-zA-Z]+@[a-zA-Z]+\\.[a-zA-Z]+$');
+
+const useEmail = (initialInput: string): useEmailOutput => {
+  const [emailState, setEmailState] = useState<string>(initialInput);
+  const [errorState, setErrorState] = useState<string | undefined>(undefined);
+
+  let errorTextComponent: TextElement = (
+    <Text style={Styles.emailErrorTextStyle}>{errorState}</Text>
```

Comment:
Is it from the task? I suppose it's better to create a component from showing an error message and use it in components where it needs.